sdks/python/src/opik/evaluation/metrics/heuristics/bleu.py

```diff
@@ -0,0 +1,218 @@
+
+    def __init__(
+        self,
+        name: str = "bleu_metric",
+        track: bool = True,
+        n_grams: int = 4,
+        smoothing_method: str = "method1",
+        weights: Optional[List[float]] = None,
+    ):
+        """
+        :param name: Name for this metric instance.
+        :param track: Whether or not this metric is tracked (depends on your system).
+        :param n_grams: Up to which n-gram order to use (1 through n_grams).
+        :param smoothing_method: One of NLTK's SmoothingFunction methods (e.g., "method0", "method1", "method2", etc.).
+        :param weights: Optional manual weighting for n-gram orders. If None, defaults to uniform across n_grams.
+        """
+        super().__init__(name=name, track=track)
+
+        # Ensure nltk is installed; if not, raise an ImportError now.
+        if nltk is None:
+            raise ImportError(
+                "`nltk` library is required for BLEU score calculation. "
+                "Please install it via `pip install nltk`."
+            )
+
+        self.n_grams = n_grams
+        self.smoothing_method = smoothing_method
+
+        # Set up weights: if not provided, default to uniform among the up to n_grams orders
+        if weights is None:
+            self.weights = [1.0 / n_grams] * n_grams
+        else:
+            if len(weights) != n_grams:
+                raise ValueError(
+                    f"Length of weights ({len(weights)}) != n_grams ({n_grams})"
+                )
+            if abs(sum(weights) - 1.0) > 1e-6:
+                raise ValueError("Weights must sum to 1.0")
+            self.weights = weights
+
+        self._nltk_smoother = SmoothingFunction()
+
+    def _get_smoothing_func(self):
+        """
+        Retrieve the corresponding smoothing function from nltk's SmoothingFunction
+        based on the self.smoothing_method name, e.g. "method0", "method1", etc.
+        Fallback to method0 if not found.
+        """
+        return getattr(self._nltk_smoother, self.smoothing_method, self._nltk_smoother.method0)
+
+    def _truncate_weights(self, candidate_len: int) -> tuple:
+        """
+        Truncate the n-gram weights to min(self.n_grams, candidate_len),
+        then re-normalize them so that they sum to 1.0.
+        """
+        max_order = min(self.n_grams, candidate_len)
+        used_weights = self.weights[:max_order]
+        w_sum = sum(used_weights) or 1.0
+        # Re-normalize to sum to 1.0
+        normalized = [w / w_sum for w in used_weights]
+        return tuple(normalized)
+
+    ###########################################################################
+    # SINGLE-SENTENCE BLEU
+    ###########################################################################
+    def score(
+        self, output: str, reference: Union[str, List[str]], **ignored_kwargs: Any
+    ) -> score_result.ScoreResult:
+        """
+        Computes a single-sentence BLEU score using nltk.translate.bleu_score.sentence_bleu.
```

Comment:
It's a public api method, please follow the same docstring format as in our other metrics.